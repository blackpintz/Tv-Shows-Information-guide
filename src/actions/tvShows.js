const addTvShowData = data => ({
  type: 'ADD_TVSHOW_DATA',
  data,
});

export default addTvShowData;
