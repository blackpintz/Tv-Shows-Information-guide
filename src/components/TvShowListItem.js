import React from 'react';
import { Link } from 'react-router-dom';
import propTypes from 'prop-types';

const TvShowListItem = ({ show }) => (
  <>
    <h4>{show.show.name}</h4>
    <Link to={`/show/${show.id}`}>
      <button type="button">See more about TV show</button>
    </Link>
  </>
);

TvShowListItem.propTypes = {
  show: propTypes.objectOf(propTypes.oneOfType([
    propTypes.string,
    propTypes.number,
    propTypes.array,
    propTypes.object,
  ])),
};

TvShowListItem.defaultProps = {
  show: 'no value',
};

export default TvShowListItem;
