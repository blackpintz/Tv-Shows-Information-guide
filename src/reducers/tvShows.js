const tvShowsReducer = (state = [], action) => {
  switch (action.type) {
    case 'ADD_TVSHOW_DATA':
      return [
        ...state, action.data,
      ];
    default:
      return state;
  }
};

export default tvShowsReducer;
