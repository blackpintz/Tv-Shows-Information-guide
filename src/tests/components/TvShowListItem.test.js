import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import TvShowListItem from '../../components/TvShowListItem';
import tvShows from '../fixtures/tvShows';

Enzyme.configure({ adapter: new Adapter() });

test('should render TvShowListItem component with the list of tvShows', () => {
  const wrapper = shallow(<TvShowListItem show={tvShows[0]} />);
  expect(wrapper).toMatchSnapshot();
});
