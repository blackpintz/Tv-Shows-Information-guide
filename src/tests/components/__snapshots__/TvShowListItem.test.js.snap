// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TvShowListItem component with the list of tvShows 1`] = `
<Fragment>
  <Col
    className="mb-3"
    lg={4}
  >
    <img
      alt="show-img"
      className="img"
      src="http://static.tvmaze.com/uploads/images/medium_portrait/105/264401.jpg"
    />
    <h5 />
    <Link
      to="/show/1"
    >
      See more
    </Link>
  </Col>
</Fragment>
`;
